native/lambda_game_engine/src/myra_engine/player.rs

```diff
@@ -0,0 +1,462 @@
+use super::utils::RelativePosition;
+use rand::Rng;
+use rustler::NifStruct;
+use rustler::NifUnitEnum;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy, NifStruct)]
+#[module = "LambdaGameEngine.MyrraEngine.Player"]
+pub struct EffectData {
+    pub time_left: MillisTime,
+    pub ends_at: MillisTime,
+    pub duration: MillisTime,
+    pub direction: Option<RelativePosition>,
+    pub position: Option<Position>,
+    pub triggered_at: MillisTime,
+    pub caused_by: u64,
+    pub caused_to: u64,
+    pub damage: u32,
+}
+
+pub type StatusEffects = HashMap<Effect, EffectData>;
+
+#[derive(rustler::NifTaggedEnum, Debug, Hash, Clone, Copy, PartialEq, Eq)]
+pub enum Effect {
+    Petrified,
+    Disarmed,
+    DenialOfService,
+    Raged,
+    NeonCrashing,
+    Leaping,
+    OutOfArea,
+    ElnarMark,
+    YugenMark,
+    XandaMark,
+    XandaMarkOwner,
+    Poisoned,
+    Slowed,
+    FieryRampage,
+    Burned,
+    Scherzo,
+    DanseMacabre,
+    Paralyzed,
+}
+impl Effect {
+    pub fn is_crowd_control(&self) -> bool {
+        match self {
+            Effect::Petrified | Effect::Disarmed => true,
+            _ => false,
+        }
+    }
+}
+
+/*
+    Note: To track cooldowns we are storing the last system time when the ability/attack
+    was used. This is not ideal, because system time is unreliable, but storing an `Instant`
+    as a field on players does not work because it can't be encoded between Elixir and Rust.
+*/
+
+#[derive(Debug, Clone, NifStruct)]
+#[module = "LambdaGameEngine.MyrraEngine.Player"]
+pub struct Player {
+    pub id: u64,
+    pub health: i64,
+    pub position: Position,
+    pub status: Status,
+    pub character: Character,
+    pub action: PlayerAction,
+    pub aoe_position: Position,
+    pub kill_count: u64,
+    pub death_count: u64,
+    // How many seconds are left until the
+    // cooldown is over.
+    pub basic_skill_cooldown_left: MillisTime,
+    pub skill_1_cooldown_left: MillisTime,
+    pub skill_2_cooldown_left: MillisTime,
+    pub skill_3_cooldown_left: MillisTime,
+    pub skill_4_cooldown_left: MillisTime,
+    // Timestamp when the cooldown started.
+    pub basic_skill_started_at: MillisTime,
+    pub skill_1_started_at: MillisTime,
+    pub skill_2_started_at: MillisTime,
+    pub skill_3_started_at: MillisTime,
+    pub skill_4_started_at: MillisTime,
+    pub basic_skill_ends_at: MillisTime,
+    pub skill_1_ends_at: MillisTime,
+    pub skill_2_ends_at: MillisTime,
+    pub skill_3_ends_at: MillisTime,
+    pub skill_4_ends_at: MillisTime,
+    // This field is redundant given that
+    // we have the Character filed, this his
+    // hopefully temporary and to tell
+    // the client which character is being used.
+    pub character_name: String,
+    pub effects: StatusEffects,
+    pub direction: RelativePosition,
+    pub body_size: f64,
+}
+
+#[derive(Debug, Clone, NifUnitEnum)]
+pub enum Status {
+    ALIVE,
+    DEAD,
+    DISCONNECTED,
+}
```

Comment:
```suggestion
pub enum Status {
    Alive,
    Dead,
    Disconnected,
}
```

Reply:
If you accept this suggestion, please remember to change it everywhere these values are used.